frontend/common/src/helpers/localize.ts

```diff
@@ -19,8 +20,13 @@ export function oppositeLocale(locale: Locales): Locales {
 }
 
 export function localizePath(path: string, locale: string): string {
-  const pathIsAbsolute = path.startsWith("/");
-  const pathSegments = path.split("/");
+
+  const {pathname, search, hash} = parsePath(path);
```

Comment:
What do you think about passing a History.Location object in instead of a string, since we're doing the transformation on both sides anyway.

Reply:
I originally left it that way because I wanted to make a lightweight patch and didn't want to change function signatures.  You're right that we do a transform in and then another out so I tried it out.  Unfortunately you end up with this gnarly createPath-localizePath-parsePath pattern in places where you do want to work with strings like Header.tsx and localize.test.ts.  I feel like you're leaking implementation details out and it doesn't really simplify much else this way.  I left in in this commit but I'd like to put it back.

Reply:
Or best of both worlds.  Leave the function taking and returning a Path but create an optional wrapper function to take and return a string that handles the transformations in just one spot?

Reply:
I see your point on the usefulness of just accepting a string, especially since we usually want a string in the end anyway. Yeah, I like your suggestion of putting it back, and using a wrapper function instead which does the common task of calling `createPath` first.

Reply:
Nice, I like this better.  I would have liked to name the wrapper with the same name but I'm getting a `Duplicate function implementation` error.  It seems like Typescript supports exporting overloaded functions but I can't get it to work.

Reply:
Huh! I did not know Typescript supported any kind of overloading. I just checkout out the [documentation](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads). I'm not sure how I feel about it? It seems like the TS implementation can get a bit messy.

They also note `Always prefer parameters with union types instead of overloads when possible`. Perhaps the best approach, in this case, would be `localizePath(path: string | Partial<Path>, locale: string): string`

Reply:
Oh yeah, the union type parameter is a way nicer solution.  The purist in me would like a version of the function that returns a `Partial<Path>` but the only place we would use that is the `redirect` call which also accepts a `string | Partial<Path>`. 😁 I like this version a lot.